drivers/usb/device/usb_dc_mcux_nxp.h

```diff
@@ -0,0 +1,372 @@
+#endif
+
+/* IP3511 FS instance count */
+#ifdef CONFIG_USB_DC_NXP_IP3511FS
+    #define USB_DEVICE_CONFIG_LPCIP3511FS (1U)
+#else
+    #define USB_DEVICE_CONFIG_LPCIP3511FS (0U)
+#endif
+
+/* IP3511 HS instance count */
+#ifdef CONFIG_USB_DC_NXP_IP3511HS
+    #define USB_DEVICE_CONFIG_LPCIP3511HS (1U)
+#else
+    #define USB_DEVICE_CONFIG_LPCIP3511HS (0U)
+#endif
+
+#define SETUP_DATA_STAGE_DONE (0)
+#define SETUP_DATA_STAGE_IN (1)
+#define SETUP_DATA_STAGE_OUT (2)
+
+/* Macro to define controller handle */
+typedef void *usb_device_handle;
+#define usb_device_controller_handle usb_device_handle
+
+/* controller driver do the ZLP for controler transfer automatically or not */
+#define USB_DEVICE_CONTROLLER_AUTO_CONTROL_TRANSFER_ZLP (0)
+
+/* endpoint related macros */
+#define EP0_MAX_PACKET_SIZE    64
+#define EP0_OUT 0
+#define EP0_IN  0x80
+/* USB endpoint mask */
+#define USB_ENDPOINT_NUMBER_MASK (0x0FU)
+/* The setup packet size of USB control transfer. */
+#define USB_SETUP_PACKET_SIZE (8U)
+/* Then endpoint number/index calculation */
+#define EP_ADDR2IDX(ep)     ((ep) & ~USB_EP_DIR_MASK)
+#define EP_ADDR2DIR(ep)     ((ep) & USB_EP_DIR_MASK)
+#define EP_ABS_IDX(ep)   (((ep) & ~USB_EP_DIR_MASK) * 2 + (((ep) & USB_EP_DIR_MASK) >> 7))
+
+/* enter critical macros */
+#define USB_OSA_SR_ALLOC() int usbOsaCurrentSr
+#define USB_OSA_ENTER_CRITICAL() usbOsaCurrentSr = irq_lock()
+#define USB_OSA_EXIT_CRITICAL() irq_unlock(usbOsaCurrentSr)
+
+#if defined(CONFIG_BIG_ENDIAN)
+#define SWAP2BYTE_CONST(n) ((((n) & 0x00FFU) << 8U) | (((n) & 0xFF00U) >> 8U))
+#define SWAP4BYTE_CONST(n) ((((n) & 0x000000FFU) << 24U) | (((n) & 0x0000FF00U) << 8U) | (((n) & 0x00FF0000U) >> 8U) | (((n) & 0xFF000000U) >> 24U))
+#define USB_SHORT_TO_LITTLE_ENDIAN(n) SWAP2BYTE_CONST(n)
+#define USB_LONG_TO_LITTLE_ENDIAN(n) SWAP4BYTE_CONST(n)
+#define USB_SHORT_FROM_LITTLE_ENDIAN(n) SWAP2BYTE_CONST(n)
+#define USB_LONG_FROM_LITTLE_ENDIAN(n) SWAP4BYTE_CONST(n)
+#else
+#define USB_SHORT_TO_LITTLE_ENDIAN(n) (n)
+#define USB_LONG_TO_LITTLE_ENDIAN(n) (n)
+#define USB_SHORT_FROM_LITTLE_ENDIAN(n) (n)
+#define USB_LONG_FROM_LITTLE_ENDIAN(n) (n)
+#endif
+
+/* Control endpoint index */
+#define USB_CONTROL_ENDPOINT (0U)
+
+/* Default invalid value or the endpoint callback length of cancelled transfer*/
+#define USB_UNINITIALIZED_VAL_32 (0xFFFFFFFFU)
+
+/* NXP SDK USB controller driver configuration macros */
+#define USB_BDT
+#define USB_GLOBAL
+#define USB_DATA_ALIGN_SIZE 4
+#define USB_RAM_ADDRESS_ALIGNMENT(n) __aligned(n)
+
+/* EHCI */
+#define USB_CONTROLLER_DATA
+/* How many the DTD are supported. */
+#define USB_DEVICE_CONFIG_EHCI_MAX_DTD (16U)
+/* Control endpoint maxPacketSize */
+#define USB_CONTROL_MAX_PACKET_SIZE (64U)
+
+/* Whether device is self power. 1U supported, 0U not supported */
+#define USB_DEVICE_CONFIG_SELF_POWER (1U)
+
+/* USB controller ID */
+typedef enum _usb_controller_index {
+	/* KHCI 0U */
+	kUSB_ControllerKhci0            = 0U,
```

Comment:
These look like MCUXpresso SDK enums. Should these defined somewhere in `ext/hal/nxp/mcux`?

Reply:
@MaureenHelm thanks, I have updated it.